### PR TITLE
Add test for meta: end_play

### PR DESCRIPTION
**Pull request by samdoran**
##### SUMMARY

We recently discoved we do not have any integration tests for `meta: end_play` feature, so I added them.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

`test/integration/targets/meta_tasks/`

### CODE REVIEW

File: test/integration/targets/meta_tasks/runme.sh [hunk 1]
```diff
@@ -14,10 +14,17 @@ done
...
   grep -q "META: ending play for testhost2" <<< "$out"
   grep -qv "play not ended for testhost" <<< "$out"
   grep -qv "play not ended for testhost2" <<< "$out"
 done
+
+# test end_play meta task
+for test_strategy in linear free; do
+  out="$(ansible-playbook test_end_play.yml -i inventory.yml -e test_strategy=$test_strategy -vv "$@")"
+
+  grep -q "META: ending play" <<< "$out"
```
Comment (mattclay) [inline test/integration/targets/meta_tasks/runme.sh:29]: Add a check to make sure the `Failed to end` message isn't present.

Comment (mattclay) [reply test/integration/targets/meta_tasks/runme.sh:29]: I would expect the non-zero exit code from fail to detect this, but it's probably worth being explicit about the test.

Comment (samdoran) [reply test/integration/targets/meta_tasks/runme.sh:29]: I did check and the test will report a failure if the `fail` task is triggered. I'll add this as another check, though. More checks are always better. 😄

**Review by mattclay — APPROVED** [review]

File: test/integration/targets/meta_tasks/test_end_play.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+- name: Testing end_play with strategy {{ test_strateg | default('linear') }}
```
Comment (mkrizek) [inline test/integration/targets/meta_tasks/test_end_play.yml]:
```suggestion
- name: Testing end_play with strategy {{ test_strategy | default('linear') }}
```

**Review by mkrizek — APPROVED** [review]

**Pull request merged**